docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+#### 3.6.1 Current implementation
+
+The display feature is implemented using a <code>DisplayCommand</code> class which extends the <code>Command</code> 
+class. 
+ 
+The <code>Duke</code> class first receives user input from the <code>Ui</code> object before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>DisplayCommand</code> object based on that user input.
+
+The <code>Duke</code> class then calls the <code>execute</code> method of the <code>DisplayCommand</code> object.
+This method makes a call to the <code>getTotalCost</code> method of the <code>ShoppingList</code> object to find the 
+cost of the items. It then calls the <code>getAmount</code> and <code>getRemainingBudget</code> methods of the 
+<code>Budget</code> object to find the current budget and the remaining budget. The results are then printed to console.
+
+The following sequence diagrams below show how the display feature works. Note the <code>Ui</code> class is
+omitted to emphasise the other classes:
+
+![alt text](images/Display.png)
+
+![alt text](images/Display_SD.png)
+
+#### 3.6.2 Design considerations
+##### Aspect: Data structure to support the display feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>DisplayCommand</code>
+ 
+  - Pros: Easy to add the display feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement display feature in the <code>Duke</code> class
+
+  - Pros: Will have less code to deal with as a new function is simply created in the <code>Duke</code> class
+  
+  - Cons: Handling the command under the <code>Duke</code> class results in longer methods. Thus, the code becomes 
+  harder to navigate and understand. 
+
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+### 3.7 Clear list feature
+This feature involves clearing all items in the shopping list. Remaining budget is also set to the user’s set budget.
+
+#### 3.7.1 Current implementation
+The clear list feature is implemented using a <code>ClearCommand</code> class which extends the <code>Command</code> 
+class. 
+
+The <code>Duke</code> class first receives user input from the <code>Ui</code> object before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>ClearCommand</code> object based on that user input.
+
+The <code>Duke</code> class then calls the <code>execute</code> method of the <code>ClearCommand</code> object which 
+makes another call to the <code>clearList</code> method of the <code>ShoppingList</code> object.
+
+The following sequence diagram below shows how the clear list feature works. Note the <code>Ui</code> class is
+omitted to emphasise the other classes:
+  
+![alt text](images/Clear.png)
```

Comment:
Is this supposed to be Duke class or Shoco class? haha

![image](https://user-images.githubusercontent.com/32284166/77893851-0eb3f100-72a7-11ea-9ed3-a8b9dde022e2.png)